src/librustc_trans/type_.rs

```diff
@@ -302,12 +268,21 @@ impl Type {
     pub fn from_integer(cx: &CrateContext, i: layout::Integer) -> Type {
         use rustc::ty::layout::Integer::*;
         match i {
-            I1 => Type::i1(cx),
             I8 => Type::i8(cx),
             I16 => Type::i16(cx),
             I32 => Type::i32(cx),
             I64 => Type::i64(cx),
             I128 => Type::i128(cx),
         }
     }
+
+    pub fn for_abi_align(ccx: &CrateContext, align: Align) -> Type {
```

Comment:
LLVM vector types have "natural alignment", i.e. they have the same alignment as their size - the `TargetDataLayout::vector_align` method used here reimplements the LLVM algorithm in question.

Reply:
Could you have a comment that you're creating oversize vectors but it's ok because they are never used by-value? And some sort of test that you can at least memcpy these around without the world exploding (not that I expect LLVM's bugs to be *that* simple, but a test is better than nothing)?

Reply:
Agreed that this is a ugly but pre-existing problem, so we should just remove the assertion and open an issue.